src/custom/constants/networks.ts

```diff
@@ -1,30 +1,146 @@
-import { JsonRpcProvider } from '@ethersproject/providers'
+import {
+  Network,
+  Web3Provider,
+  JsonRpcProvider,
+  UrlJsonRpcProvider,
+  StaticJsonRpcProvider,
+  InfuraProvider,
+  CloudflareProvider,
+  AlchemyProvider,
+  PocketProvider,
+  AnkrProvider,
+} from '@ethersproject/providers'
+import * as Sentry from '@sentry/react'
 
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Newable } from '@cow/types'
 
 const INFURA_KEY = process.env.REACT_APP_INFURA_KEY
 if (typeof INFURA_KEY === 'undefined') {
   throw new Error(`REACT_APP_INFURA_KEY must be a defined environment variable`)
 }
 
-export const MAINNET_PROVIDER = new JsonRpcProvider(`https://mainnet.infura.io/v3/${INFURA_KEY}`)
+interface RpcConfig extends Network {
+  // You should only need to provide this if you are trying to connect to a custom RPC / network.
+  readonly rpcUrl?: string
+}
+
+const STATIC_RPC_URLS: Partial<Record<SupportedChainId, string>> = {
+  [SupportedChainId.GNOSIS_CHAIN]: 'https://rpc.gnosis.gateway.fm',
+}
+
+export const RPC_CONFIG: Record<SupportedChainId, RpcConfig> = {
+  [SupportedChainId.MAINNET]: {
+    chainId: SupportedChainId.MAINNET,
+    name: 'homestead',
+  },
+  [SupportedChainId.GOERLI]: {
+    chainId: SupportedChainId.GOERLI,
+    name: 'goerli',
+  },
+  [SupportedChainId.GNOSIS_CHAIN]: {
+    chainId: SupportedChainId.GNOSIS_CHAIN,
+    name: 'gnosis',
+    rpcUrl: STATIC_RPC_URLS[SupportedChainId.GNOSIS_CHAIN],
+  },
+}
 
 /**
- * These are the network URLs used by the interface when there is not another available source of chain data
+ * We use this class for type inference.
+ *
+ * UrlJsonRpcProvider is an abstract class, therefore we have trouble inferring the type of its constructor.
```

Comment:
`we have trouble` or `can't`?

i would think if it's abstract, we don't know the constructor of the children. Is this what you are refering to?

Reply:
I wrote it with my downton abbey glasses apparently 😅 Will rephrase, you are correct, we can't.